presto-main/src/main/java/com/facebook/presto/server/QueryStateInfoResource.java

```diff
@@ -60,6 +61,7 @@ public QueryStateInfoResource(
         }
         List<QueryInfo> userQueries = queryManager.getAllQueryInfo().stream()
                 .filter(queryInfo -> Pattern.matches(user, queryInfo.getSession().getUser()))
+                .filter(queryInfo -> queryInfo.getState() != FINISHED)
```

Comment:
I'm not sure whether we want to remove failed queries. Might to useful? @martint What do you think?